symbolic-symcache/src/writer.rs

```diff
@@ -381,8 +381,16 @@ where
             return Ok(*index);
         }
 
-        if self.files.len() >= std::u16::MAX as usize {
-            return Err(SymCacheErrorKind::TooManyValues(ValueKind::File).into());
+        // Instead of failing hard when exceeding the maximum allowed number of files (u16::MAX),
+        // we rather insert a sentinel "<unknown>" file and use that for all files that exceed the max.
```

Comment:
Please also mark this as `TODO`, since we now clearly need that we need to support more than 65k files.